programming_examples/channel_examples/worker_to_self/worker_to_self.py

```diff
@@ -23,14 +23,18 @@ def build_module():
     ChannelOp("ChanOut")
```

Comment:
I attempted to change this file to use the new ability to send L2 allocations made in a segment to a herd. However, this example still does not work. I'm unsure of how I've rewritten it is reasonable or not, but I've made a note in the associated issue (https://github.com/Xilinx/mlir-air/issues/654)

Reply:
The `tensor_in_l2` seems to be deallocated before it was used in `air.herd`, if the code was interpreted line by line. Maybe move the deallocation to after the `herd`.

Reply:
I fixed a bug (I accidentally was using L1 for both allocation types) and made the change you suggested, as well as simplifying the example a bit. The example is still not successful, although the error changed:
```
Running: builtin.module(air-to-aie{emit-while-loop=false row-offset=2 col-offset=0 device=npu1_4col})
Segmentation fault (core dumped)
```